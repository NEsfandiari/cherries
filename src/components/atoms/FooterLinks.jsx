import React, { Component } from 'react'
import Styled from 'styled-components'
import { Link } from 'gatsby'

const Container = Styled.div`
display: flex;
flex-basis: 50%;
align-items: center;
justify-content: center;
.list-items {
  text-decoration: none;
  color: white;
}
li {
  padding-right: 3.6rem;
}
.list {
  height: 10rem;
  list-style-type: none;
  padding-top: 1rem;
  padding-right: 1rem;
  flex-direction: column;
  flex-wrap: wrap;
  display: flex;
  flex-basis: 50%;
}
`
class FooterLinks extends Component {
  render() {
    const links = this.props.linkData.map(d => (
      <li>
        <Link className="list-items" to={d.content[1].data.uri}>
          {d.content[1].content[0].value}
        </Link>
      </li>
    ))
    return (
      <Container>
        <ul className="list">{links}</ul>
      </Container>
    )
  }
}

export default FooterLinks
