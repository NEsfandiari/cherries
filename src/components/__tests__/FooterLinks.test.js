import React from 'react'
import renderer from 'react-test-renderer'
import FooterLinks from '../atoms/FooterLinks'

describe('FooterLinks', () =>
  it('renders correctly', () => {
    const linkData = [
      {
        content: [
          {
            data: {
              uri: null,
            },
            content: null,
          },
          {
            data: {
              uri: 'https://lipslut.com',
            },
            content: [
              {
                value: 'Refer a Friend',
              },
            ],
          },
          {
            data: {
              uri: null,
            },
            content: null,
          },
        ],
      },
    ]

    const tree = renderer.create(<FooterLinks linkData={linkData} />).toJSON()
    expect(tree).toMatchSnapshot()
  }))
