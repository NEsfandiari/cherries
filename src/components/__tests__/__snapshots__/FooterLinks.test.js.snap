// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterLinks renders correctly 1`] = `
<div
  className="sc-bdVaJa jYeLTA"
>
  <ul
    className="list"
  >
    <li>
      <a
        className="list-items"
        href="https://lipslut.com"
      >
        Refer a Friend
      </a>
    </li>
  </ul>
</div>
`;
